### ISSUE TITLE
Fail to install 'pgenlibr' R package

### ISSUE THREAD

**Issue by richelbilderbeek** [chrchang/plink-ng#204]
Dear plink-ng maintainers,

Thanks fo `plink-ng` and the `pgenlibr` R package.

Here I want to report that the `pgenlibr` package cannot be installed installed, as I learned from [one of my GitHub Actions builds](https://github.com/richelbilderbeek/plinkr/runs/4775460101?check_suite_focus=true#step:7:1267).

To reproduce, within R do:

```r
remotes::install_github("chrchang/plink-ng/2.0/pgenlibr")
```

This results in in error, due to symbol `HUF_decompress4X1_usingDTable_internal_bmi2_asm_loop` not being found in the `.so` file (see the full error message below).

I hope `pgenlibr` will be fixed soon.

Thanks and cheers, Richel Bilderbeek

## Error

```
Using github PAT from envvar GITHUB_PAT
Downloading GitHub repo chrchang/plink-ng@HEAD
✔  checking for file ‘/tmp/RtmpNhiQCj/remotesa2fe5377086e/chrchang-plink-ng-abb78c3/2.0/pgenlibr/DESCRIPTION’ ...
─  preparing ‘pgenlibr’:
✔  checking DESCRIPTION meta-information ...
─  cleaning src
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘pgenlibr_0.3.1.tar.gz’
   
Installing package into ‘/home/richel/R/x86_64-pc-linux-gnu-library/4.1’
(as ‘lib’ is unspecified)
* installing *source* package ‘pgenlibr’ ...
** using staged installation
** libs
g++ -std=gnu++14 -I"/usr/share/R/include" -DNDEBUG -DSTATIC_ZSTD -Izstd/lib -Izstd/lib/common -I'/home/richel/R/x86_64-pc-linux-gnu-library/4.1/Rcpp/include'    -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-i2PIHO/r-base-4.1.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c include/plink2_base.cc -o include/plink2_base.o
[...]
-D_FORTIFY_SOURCE=2 -g  -c libdeflate/lib/zlib_compress.c -o libdeflate/lib/zlib_compress.o
gcc -std=gnu99 -std=gnu11 -I"/usr/share/R/include" -DNDEBUG -DSTATIC_ZSTD -Izstd/lib -Izstd/lib/common -I'/home/richel/R/x86_64-pc-linux-gnu-library/4.1/Rcpp/include'   -Izstd/lib -Izstd/lib/common -Ilibdeflate -Ilibdeflate/common -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-i2PIHO/r-base-4.1.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c libdeflate/lib/utils.c -o libdeflate/lib/utils.o
gcc -std=gnu99 -std=gnu11 -I"/usr/share/R/include" -DNDEBUG -DSTATIC_ZSTD -Izstd/lib -Izstd/lib/common -I'/home/richel/R/x86_64-pc-linux-gnu-library/4.1/Rcpp/include'   -Izstd/lib -Izstd/lib/common -Ilibdeflate -Ilibdeflate/common -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-i2PIHO/r-base-4.1.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c libdeflate/lib/gzip_decompress.c -o libdeflate/lib/gzip_decompress.o
gcc -std=gnu99 -std=gnu11 -I"/usr/share/R/include" -DNDEBUG -DSTATIC_ZSTD -Izstd/lib -Izstd/lib/common -I'/home/richel/R/x86_64-pc-linux-gnu-library/4.1/Rcpp/include'   -Izstd/lib -Izstd/lib/common -Ilibdeflate -Ilibdeflate/common -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-i2PIHO/r-base-4.1.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c libdeflate/lib/x86/x86_cpu_features.c -o libdeflate/lib/x86/x86_cpu_features.o
g++ -std=gnu++14 -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o pgenlibr.so include/plink2_base.o include/plink2_bits.o include/pgenlib_misc.o include/pgenlib_read.o pvar_ffi_support.o pgenlib_ffi_support.o include/plink2_bgzf.o include/plink2_string.o include/plink2_text.o include/plink2_thread.o include/plink2_zstfile.o pvar.o pgenlibr.o RcppExports.o zstd/lib/compress/zstd_lazy.o zstd/lib/compress/zstd_double_fast.o zstd/lib/compress/zstd_compress_superblock.o zstd/lib/compress/zstd_fast.o zstd/lib/compress/zstd_ldm.o zstd/lib/compress/hist.o zstd/lib/compress/huf_compress.o zstd/lib/compress/fse_compress.o zstd/lib/compress/zstd_compress_literals.o zstd/lib/compress/zstd_compress_sequences.o zstd/lib/compress/zstd_compress.o zstd/lib/compress/zstdmt_compress.o zstd/lib/compress/zstd_opt.o zstd/lib/decompress/huf_decompress.o zstd/lib/decompress/zstd_decompress.o zstd/lib/decompress/zstd_decompress_block.o zstd/lib/decompress/zstd_ddict.o zstd/lib/common/fse_decompress.o zstd/lib/common/entropy_common.o zstd/lib/common/pool.o zstd/lib/common/error_private.o zstd/lib/common/xxhash.o zstd/lib/common/debug.o zstd/lib/common/threading.o zstd/lib/common/zstd_common.o libdeflate/lib/crc32.o libdeflate/lib/gzip_compress.o libdeflate/lib/adler32.o libdeflate/lib/deflate_decompress.o libdeflate/lib/zlib_decompress.o libdeflate/lib/deflate_compress.o libdeflate/lib/zlib_compress.o libdeflate/lib/utils.o libdeflate/lib/gzip_decompress.o libdeflate/lib/x86/x86_cpu_features.o -lpthread -L/usr/lib/R/lib -lR
installing to /home/richel/R/x86_64-pc-linux-gnu-library/4.1/00LOCK-pgenlibr/00new/pgenlibr/libs
** R
** byte-compile and prepare package for lazy loading
** help
No man pages found in package  ‘pgenlibr’ 
*** installing help indices
** building package indices
** testing if installed package can be loaded from temporary location
Error: package or namespace load failed for ‘pgenlibr’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/home/richel/R/x86_64-pc-linux-gnu-library/4.1/00LOCK-pgenlibr/00new/pgenlibr/libs/pgenlibr.so':
  /home/richel/R/x86_64-pc-linux-gnu-library/4.1/00LOCK-pgenlibr/00new/pgenlibr/libs/pgenlibr.so: undefined symbol: HUF_decompress4X1_usingDTable_internal_bmi2_asm_loop
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/home/richel/R/x86_64-pc-linux-gnu-library/4.1/pgenlibr’
* restoring previous ‘/home/richel/R/x86_64-pc-linux-gnu-library/4.1/pgenlibr’
Warning message:
In i.p(...) :
  installation of package ‘/tmp/RtmpNhiQCj/filea2fe3a12d7e4/pgenlibr_0.3.1.tar.gz’ had non-zero exit status
```

**Comment by richelbilderbeek** [comment]
By the way, this error would have been discovered earlier would `plink-ng` have its own GitHub Actions test suite. As an experienced GitHub Actions user, I do volunteer to reproduce this error in a GitHub Actions script, so that this error is found quicker in the future.

**Comment by richelbilderbeek** [comment]
Additionally: this work fine until 15 days ago, so error must have been introduced after Dec 27th 2021, so it can only be in [one of five last commits to master](https://github.com/chrchang/plink-ng/commits/master)

**Comment by chrchang** [comment]
Oops; this should be fixed by 0f9e10b .  Yes, you're more than welcome to add a GitHub Actions test.

**Comment by richelbilderbeek** [comment]
@chrchang: Cool, I made a Pull Request at #206 . I can already verify it works, thanks for fixing it!

Feel free to close this Issue :-)
